Firebase/Database/Core/FServerValues.m

```diff
@@ -20,37 +20,118 @@
 #import "FLeafNode.h"
 #import "FSnapshotUtilities.h"
 
+const NSString *kTimestamp = @"timestamp";
+const NSString *kIncrement = @"increment";
+
+BOOL longCanHold(NSNumber *num) {
+    switch (num.objCType[0]) {
+    case 'f': // float; fallthrough
+    case 'd': // double
+        break;
+    case 'L': // unsigned long; fallthrough
+    case 'Q': // unsigned long long; fallthrough
+        // Only use ulong(long) if there isn't an overflow.
+        if (num.unsignedLongLongValue > LONG_MAX) {
+            break;
+        }
+    default:
+        return YES;
+    }
+    return NO;
```

Comment:
I think this could be more readable. What about something like this:


``` 
switch (num.objCType[0]) {
    case 'f': // float; fallthrough
    case 'd': // double
        return NO;
    case 'L': // unsigned long; fallthrough
    case 'Q': // unsigned long long; fallthrough
        // Only use ulong(long) if there isn't an overflow.
        if (num.unsignedLongLongValue > LONG_MAX) {
             return NO;;
        }
    }
    return YES;
}
```

I am mostly interested in getting rid of the "break" statements as the code flow is a little confusing.